src/db/image_repository.py

```
@@ -0,0 +1,31 @@
+from image_event import image_event_contract_pb2
+from src.db import connection
+
+def upsert_image(protobuf_message):
+    conn = connection.get_connection()
+
+    print("upserting image")
+    cursor = conn.cursor()
+    record_to_insert = (protobuf_message.upsertImageEvent.originId,
+                        protobuf_message.upsertImageEvent.originTable,
+                        protobuf_message.upsertImageEvent.s3Key,
+                        protobuf_message.upsertImageEvent.s3Bucket,
+                        image_event_contract_pb2.ImageEvent.ImageCategory.Name(protobuf_message.upsertImageEvent.imageCategory))
+
+    print("executing query: \n" + str(UPSERT_IMAGE_QUERY) + "\nwith values \n" + str(record_to_insert))
+
+    try:
+        cursor.execute(UPSERT_IMAGE_QUERY, record_to_insert)
+        conn.commit()
+    except Exception as ex:
+        print(str(ex))
+        raise
```

This is kind of meaningless, if you raise the exception it will get printed to cloudwatch, and you are just going to raise the exception anyway so you didn't catch it (and you won't hit your `finally:`, not that it matters)
You should just let it fail, or alternatively if you use the built-in python logging you can to `log.exception(ex)` and get the full exception traceback without actually raising anything.